microdnf/commands/distro-sync/distro-sync.cpp

```diff
@@ -34,25 +38,58 @@ namespace microdnf {
 using namespace libdnf::cli;
 
 
-DistroSyncCommand::DistroSyncCommand(Command & parent) : Command(parent, "distro-sync") {
+DistroSyncCommand::DistroSyncCommand(Command & parent) : Command(parent, "distro-sync") {}
+
+DistroSyncCommand::DistroSyncCommand(Command & parent, const std::string & name) : Command(parent, name) {
     auto & ctx = static_cast<Context &>(get_session());
     auto & parser = ctx.get_argument_parser();
 
     auto & cmd = *get_argument_parser_command();
     cmd.set_short_description("Upgrade or downgrade installed software to the latest available versions");
 
-    patterns = parser.add_new_values();
+    patterns_to_distro_sync_options = parser.add_new_values();
     auto patterns_arg = parser.add_new_positional_arg(
         "patterns",
         ArgumentParser::PositionalArg::UNLIMITED,
         parser.add_init_value(std::unique_ptr<libdnf::Option>(new libdnf::OptionString(nullptr))),
-        patterns);
+        patterns_to_distro_sync_options);
     patterns_arg->set_short_description("Patterns");
     cmd.register_positional_arg(patterns_arg);
 }
 
 
 void DistroSyncCommand::run() {
+    auto & ctx = static_cast<Context &>(get_session());
+
+    ctx.load_repos(true);
+
+    std::cout << std::endl;
+
+    libdnf::Goal goal(ctx.base);
+    if (patterns_to_distro_sync_options->empty()) {
```

Comment:
@m-blaha this line currently segfaults. I cannot quite understand why. Any ideas?

Reply:
The problem is not here, but on the line 41. Instead of calling another DistroSyncCommand constructor with name set to "distro-sync", you call parent class (`Command`) constructor. As a result `DistroSyncCommand` constructor defined on lines 43-58 is not called, therefore the `patterns_to_distro_sync_options` class member is not initialized (or more precisely is initialized to nullptr) and when you try to access it, the program segfaults.
The line 41 should be:
```
DistroSyncCommand::DistroSyncCommand(Command & parent) : DistroSyncCommand(parent, "dis  tro-sync") {}
```

Reply:
oh! that's right. thank you. now should be done